packages/sitecore-jss-nextjs/src/edge/sitemap-middleware.ts

```diff
@@ -0,0 +1,65 @@
+import { NextResponse, NextRequest } from 'next/server';
+import {
+  GraphQLSitemapService,
+  GraphQLSitemapServiceConfig,
+} from '@sitecore-jss/sitecore-jss/site';
+
+/**
+ * extended SitemapMiddlewareConfig config type for SitemapMiddleware
+ */
+export type SitemapMiddlewareConfig = Omit<GraphQLSitemapServiceConfig, 'fetch'>;
+
+/**
+ * Middleware / handler fetches all sitemap from Sitecore instance by grapqhl service
+ * compares with current url and sitemap to target url
+ */
+export class SitemapMiddleware {
+  private sitemapService: GraphQLSitemapService;
+
+  /**
+   * NOTE: we provide native fetch for compatibility on Next.js Edge Runtime
+   * (underlying default 'cross-fetch' is not currently compatible: https://github.com/lquixada/cross-fetch/issues/78)
+   * @param {SitemapMiddlewareConfig} [config] sitemap middleware config
+   */
+  constructor(private config: SitemapMiddlewareConfig) {
+    this.sitemapService = new GraphQLSitemapService({ ...this.config, fetch: fetch });
+  }
+
+  /**
+   * Gets the Next.js API route handler
+   * @returns route handler
+   */
+  public getHandler(req: NextRequest): Promise<NextResponse> {
+    return this.handler(req);
+  }
+
+  /**
+   * handler which fetc sitemaps and rewrite url
+   * @param req<NextRequest>
+   * @returns Promise{NextResponse}
+   */
+  private handler = async (req: NextRequest): Promise<NextResponse> => {
+    const url = req.nextUrl.clone();
+    // Find the sitemap from result of SitemapService
+    const existsSitemap = await this.getExistsSitemap(url);
+
+    if (existsSitemap) {
+      const url = `${this.config.sitecoreApiHost}${existsSitemap}`;
```

Comment:
**Read this comment first :) Some of the others may be moot**

The sitemap feature doesn't need to be on Next.js edge middleware, it should just be an API route with redirect (similar to robots feature). Also wondering if we need a dedicated middleware. There is not much logic here, we can probably just add the code using `GraphQLSitemapService` directly to the API route (again, like robots). Maybe build the `getExistsSitemap` into the `GraphQLSitemapService` itself.

Also, the sitecoreApiHost is actually [not required (and shouldn't be used) on Edge](https://doc.sitecore.com/xp/en/developers/hd/190/sitecore-headless-development/walkthrough--deploying-jss-next-js-apps-to-vercel.html). We need to get rid of this dependency in the config. The move to an API route hopefully takes care of this?